code/modules/mining/lavaland/loot/ashdragon_loot.dm

```diff
@@ -191,6 +192,14 @@
 	if(is_type_in_typecache(target, banned_turfs))
 		return
 
+	if(!is_mining_level(user.z)) //Will only spawn a few sparks if not on mining z level
+
+		if(world.time > timerMalfunction)
+			timerMalfunction = world.time + create_delay + 1
+			user.visible_message("<span class='danger'>[user]'s lava staff malfunctions!</span>")
+			do_sparks(5, FALSE, user)
+		return
+
```

Comment:
Styling suggestion using an early return instead. If you don't decide to go with this, I still suggest removing the trailing `return`, as it is unnecessary. 
```suggestion
	if(is_mining_level(user.z)) //Will only spawn a few sparks if not on mining z level
	    return

	if(world.time > timerMalfunction)
		timerMalfunction = world.time + create_delay + 1
		user.visible_message("<span class='danger'>[user]'s lava staff malfunctions!</span>")
		do_sparks(5, FALSE, user)

```

Reply:
Your suggested change would prevent the staff from working at all on the mining levels

Reply:
You're right I was tunnel visioned on this one code snippet, didn't think about the rest of the proc. Ignore what I said.